source/dub/commandline.d

```diff
@@ -678,7 +678,6 @@ class CommandArgs {
 	{
 		foreach (ref arg; m_recognizedArgs)
 			if (names == arg.names) {
-				assert(help_text is null);
```

Comment:
don't remove this assert, it checks if there are duplicate recognized args, and if there are the duplicates must not change the help text.

Instead you want to remove the `temp-build` getopt from RunCommand, as you now added it in BuildCommand, which RunCommand inherits from

Reply:
thanks for the explanation. i reworked the commits and added a perhaps clearer message to the error.